dwave/optimization/include/dwave-optimization/array.hpp

```diff
@@ -935,6 +935,10 @@ std::ostream& operator<<(std::ostream& os, const Array::View& view);
 bool array_shape_equal(const Array* lhs_ptr, const Array* rhs_ptr);
 bool array_shape_equal(const Array& lhs, const Array& rhs);
 
+// Test whether multiple arrays all have the same shape.
+bool array_shape_equal(const std::span<const Array* const> array_ptrs);
+bool array_shape_equal(const std::vector<const Array*>& array_ptrs);
```

Comment:
Isn't this overload redundant? Do you mean to say
```c++
bool array_shape_equal(const std::vector<const Array*> array_ptrs);
```
or
```c++
bool array_shape_equal(const std::vector<const Array*>&& array_ptrs);
```
?

I am guessing we can just drop it.

Reply:
You're right, probably leftover from when I was trying to get the rvalue version to work.